src/test/kotlin/com/pubnub/api/crypto/CryptoModuleTest.kt

```diff
@@ -0,0 +1,143 @@
+        val msgToEncrypt = "Hello world".toByteArray()
+
+        // when
+        val encryptedMsg = legacyCryptoModule.encrypt(msgToEncrypt)
+        val decryptedMsg = legacyCryptoModule.decrypt(encryptedMsg)
+
+        // then
+        Assert.assertTrue(msgToEncrypt.contentEquals(decryptedMsg))
+    }
+
+    @Test
+    fun `using LegacyCryptoModule can decrypt message that was encrypted with AesCbcCryptor`() {
+        // given
+        val cipherKey = "enigma"
+        val moduleWithAesCbcCryptorOnly = CryptoModule.createNewCryptoModule(defaultCryptor = AesCbcCryptor(cipherKey))
+        val legacyCryptoModule = CryptoModule.createLegacyCryptoModule(cipherKey)
+        val msgToEncrypt = "Hello world".toByteArray()
+
+        // when
+        val encryptedMsg = moduleWithAesCbcCryptorOnly.encrypt(msgToEncrypt)
+        val decryptedMsg = legacyCryptoModule.decrypt(encryptedMsg)
+
+        // then
+        Assert.assertTrue(msgToEncrypt.contentEquals(decryptedMsg))
+    }
+
+    @Test
+    fun `using AesCbcCryptoModule can decrypt message that was encrypted with LegacyCryptor with randomIV `() {
+        // given
+        val cipherKey = "enigma"
+        val moduleWithLegacyCryptorOnlyWithRandomIV = CryptoModule.createNewCryptoModule(defaultCryptor = LegacyCryptor(cipherKey))
+        val aesCbcCryptoModule = CryptoModule.createAesCbcCryptoModule(cipherKey)
+        val msgToEncrypt = "Hello world".toByteArray()
+
+        // when
+        val encryptedMsg = moduleWithLegacyCryptorOnlyWithRandomIV.encrypt(msgToEncrypt)
+        val decryptedMsg = aesCbcCryptoModule.decrypt(encryptedMsg)
+
+        // then
+        Assert.assertTrue(msgToEncrypt.contentEquals(decryptedMsg))
+    }
+
+    @Test
+    fun `using AesCbcCryptoModule can decrypt message that was encrypted with LegacyCryptor with staticIV `() {
+        // given
+        val cipherKey = "enigma"
+        val moduleWithLegacyCryptorOnlyWithStaticIV = CryptoModule.createNewCryptoModule(defaultCryptor = LegacyCryptor(cipherKey, false))
+        val aesCbcCryptoModule = CryptoModule.createAesCbcCryptoModule(cipherKey, false)
+        val msgToEncrypt = "Hello world".toByteArray()
+
+        // when
+        val encryptedMsg = moduleWithLegacyCryptorOnlyWithStaticIV.encrypt(msgToEncrypt)
+        val decryptedMsg = aesCbcCryptoModule.decrypt(encryptedMsg)
+
+        // then
+        Assert.assertTrue(msgToEncrypt.contentEquals(decryptedMsg))
+    }
+
+    @Test
+    fun `using module wIth only AesCbcCryptor can decrypt encrypted message`() {
+        // given
+        val cipherKey = "enigma"
+        val aesCbcCryptoModule = CryptoModule.createAesCbcCryptoModule(cipherKey,)
+        val msgToEncrypt = "Hello world".toByteArray()
+
+        // when
+        val encryptedMsg = aesCbcCryptoModule.encrypt(msgToEncrypt)
+        val decryptedMsg = aesCbcCryptoModule.decrypt(encryptedMsg)
+
+        // then
+        Assert.assertTrue(msgToEncrypt.contentEquals(decryptedMsg))
```

Comment:
I think it's better to use `assertArrayEquals` instead of `assertTrue` when comparing arrays. Almost everything is better than `assertTrue`. The more specific assert the more information you'll get when failing

Reply:
Ok. I will change it.